### ISSUE TITLE
Download missing ROMs yields "forbidden"

### ISSUE THREAD

**Issue by a2geek** [ksherlock/ample#19]
When I try to download the ROMs via the ROM manager, it shows `forbidden`. The original URL didn't work so I had to use `https://archive.org/download/mame0.224` because I saw it while poking around in the code.

Uncertain if this is useful, but I got 302's via `curl` due to the `Location:` header.

What I see in the interface is this:

<img width="465" alt="image" src="https://user-images.githubusercontent.com/11591819/140674923-b9173100-814d-4c79-bfdc-e3218e86946f.png">

What I ended up doing is just a shell hack (in case this helps others):
```
curl --silent https://raw.githubusercontent.com/ksherlock/ample/345d4f3d740c213aac47e59114466c80809e171d/Ample/Resources/roms.plist | grep "<string>" | sed 's/<string>//' | sed 's#</string>##' | tr -d ' ' | xargs -t -I {} -n 1 curl --silent -LO https://archive.org/download/mame0.224/{}.zip
```

**Comment by ksherlock** [comment]
archive.org mame rom uploads have a tendency to randomly disappear. Since it's so inconsistent I assume the original uploader deleted it.  

"https://archive.org/download/mame-0.231-merged" with a filetype of 7z might be the most up to date one I see, as of today.

**Comment by a2geek** [comment]
Tried that as well but Ample still gives me a Forbidden error.
<img width="465" alt="image" src="https://user-images.githubusercontent.com/11591819/141033274-ad2247b7-2292-4297-99d5-9134b2a10c98.png">

The ROMs I downloaded by hand and they work in Ample just fine. Just having issues with the automated download piece.

**Comment by ksherlock** [comment]
That's very peculiar.  The 0.231 and 0.224 URLs both download succesfully for me.

The console log might have more details.

**Comment by a2geek** [comment]
I launched from the command-line. . . . and it worked. I then deleted the items launched from the desktop and it worked again. No clue what was going on before since I can't replicate what was happening earlier this evening! Is there anything I should look at that may help out?  If not, feel free to close. I can't explain it and don't want to waste time if it's just me.  I'm on the M1 Pro if that's useful.

**Comment by rharke** [comment]
I also cannot download any ROMs.

Tried: https://archive.org/download/mame-0.231-merged and https://archive.org/details/mame-merged

I just get "forbidden" from all of them.

**Comment by ksherlock** [comment]
Thanks both of you.  I found the issue - The internal URL isn't updated until the URL text field loses focus.  For now, press `tab` after changing the URL.  The next update will handle it better.

**Comment by ksherlock** [comment]
Fixed in Ample 0.238
